app/api/admin_statistics_api/events.py

```diff
@@ -50,3 +53,405 @@ class AdminStatisticsEventDetail(ResourceDetail):
         'class': NoModelLayer,
         'session': db.session
     }
+
+
+class AdminStatisticsEventTypeSchema(Schema):
+    """
+    Api schema
+    """
+    class Meta:
+        """
+        Meta class
+        """
+        type_ = 'admin-statistics-event-type'
+        self_view = 'v1.admin_statistics_event_type_detail'
+        inflect = dasherize
+
+    id = fields.String()
+    camp_treat_retreat = fields.Method("camp_treat_retreat_count")
+    appearance_or_signing = fields.Method("appearance_or_signing_count")
+    attraction = fields.Method("attraction_count")
+    class_training_or_workshop = fields.Method("class_training_or_workshop_count")
+    concert_or_performance = fields.Method("concert_or_performance_count")
+    conference = fields.Method("conference_count")
+    convention = fields.Method("convention_count")
+    dinner_or_gala = fields.Method("dinner_or_gala_count")
+
+    festival_or_fair = fields.Method("festival_or_fair_count")
+    game_or_competition = fields.Method("game_or_competition_count")
+    meeting_or_networking_event = fields.Method("meeting_or_networking_event_count")
+    party_or_social_gathering = fields.Method("party_or_social_gathering_count")
+    race_or_endurance_event = fields.Method("race_or_endurance_event_count")
+
+    rally = fields.Method("rally_count")
+    screening = fields.Method("screening_count")
+    seminar_or_talk = fields.Method("seminar_or_talk_count")
+    tour = fields.Method("tour_count")
+    tournament = fields.Method("tournament_count")
+    other = fields.Method("other_count")
+
+    @staticmethod
+    def camp_treat_retreat_count(obj):
+        try:
+            events_type = EventType.query.filter_by(name='Camp, Treat & Retreat').one()
+        except NoResultFound:
+            return 0
+        return get_count(Event.query.filter_by(event_type_id=events_type.id))
+
+    @staticmethod
+    def appearance_or_signing_count(obj):
```

Comment:
Its because marshmallow provides Method field `fields.Method()` which does not contain any arguments other then obj 
Please see under Method Fields section at http://marshmallow.readthedocs.io/en/latest/custom_fields.html

Reply:
[Example](http://marshmallow.readthedocs.io/en/latest/custom_fields.html) provided is returning whether a user is author the blog or not .
In our case, 
```
define function as 
def get_event_type_count(event_type_name):
    try:
        events_type = EventType.query.filter_by(name=event_type_name).one()
    except NoResultFound:
        return 0
    return get_count(Event.query.filter_by(event_type_id=events_type.id))

c_or_p =  get_event_type_count('Concert or Performance')
attr =  get_event_type_count('Attraction')

schema.context = {'Concert or Performance': c_or_p, 'Attraction': attr, and so on}

**In Schemas attributes**

concert_or_performance = fields.Function(lambda value, context: value == context['Concert or Performance':])
 
```

Not sure if it's write but expected @schedutron

Reply:
I think it's very simple now, create a helper function in helpers and import it here. Now call it from lamda function like:
`concert_or_performance = fields.Function(lambda value: stats_helper_func('Concert or Performance'))`
 
I think this will do

Reply:
Please review @poush